### DISCUSSION TITLE
Decimal data type is not able to store the large number above 1e+20

### DISCUSSION THREAD

**Discussion by moniratna** [prisma/prisma#16717] [Q&A]
I am trying to store 1000000000000000000000 value using new Prisma.Decimal("1000000000000000000000").but it is not storing as it is.
It is storing the value 1000000000000000 and it lost 3 precisions.

Can someone help me on this issue?

**Comment by sakshamAtNord** [comment]
## Title: 
When storing `1000000000000000000000` using new Prisma.Decimal(amount), the value is losing 3 precision values and is being stored as `1000000000000000`.

## Description:
We are using Prisma as a ORM with PostgreSQL in our express app to store user balance. 

Following is the code which we are using:

```
const updateBalance = await prismaClient.userAssets.update({
    where: {
        userId_assetId: {
            userId: userId,
            assetId: assetId,
        },
    },
    data: {
        availableBalance: new Prisma.Decimal("1000000000000000000000"),
    },
});
```

When we simply console log the availableBalance it is coming as: `1e+21`.
While in DB it is being stored as `1000000000000000`


## Context:
Schema of availableBalance in userAssets: `availableBalance Decimal      @default(0) @db.Decimal(36, 0)`
Prisma client version: "@prisma/client": "^3.15.1",
PostgreSQL version: 
Express version: "express": "^4.18.1",
NodeJS version: v14.21.1
OS version: Ubuntu 20.04 hosted on AWS EC2

**Comment by sakshamAtNord** [comment]
Hey @SevInf , could you please help with this?

**Comment by SevInf** [comment]
@sakshamAtNord could you try with more recent Prisma version? There have been a bunch of fixes around decimal precision since 3.15.1

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
